pennylane/ops/qubit/hamiltonian.py

```diff
@@ -616,8 +616,16 @@ def compare(self, other):
         if isinstance(other, Hamiltonian):
             self.simplify()
             other.simplify()
-            return self._obs_data() == other._obs_data()  # pylint: disable=protected-access
 
+            for selfdata in self._obs_data():
+                flag = False
```

Comment:
A slightly more descriptive name for this boolean would probably be more appropriate, maybe something like `is_equal`?